pallets/issue/src/tests.rs

```diff
@@ -624,3 +624,643 @@ fn test_request_issue_reset_interval_and_succeeds_with_rate_limit() {
+
+			let expected_value_vault_1_fee_1: u128 =
+				((vault_1_collateral_usd as f64 / total_amount_usd as f64) * issue_fee_1 as f64)
+					.floor() as u128;
+
+			let expected_value_vault_2_fee_1: u128 =
+				((vault_2_collateral_usd as f64 / total_amount_usd as f64) * issue_fee_1 as f64)
+					.floor() as u128;
+
+			let expected_value_vault_1_fee_2: u128 =
+				((vault_1_collateral_usd as f64 / total_amount_usd as f64) * issue_fee_2 as f64)
+					.floor() as u128;
+
+			let expected_value_vault_2_fee_2: u128 =
+				((vault_2_collateral_usd as f64 / total_amount_usd as f64) * issue_fee_2 as f64)
+					.floor() as u128;
+
+			assert_eq!(get_reward_for_vault(&VAULT, issue_asset_1), expected_value_vault_1_fee_1);
+			assert_eq!(get_reward_for_vault(&VAULT_2, issue_asset_1), expected_value_vault_2_fee_1);
+
+			assert_eq!(get_reward_for_vault(&VAULT, issue_asset_2), expected_value_vault_1_fee_2);
+			assert_eq!(get_reward_for_vault(&VAULT_2, issue_asset_2), expected_value_vault_2_fee_2);
+		})
+	}
+
+	fn execute_multiple_issues() -> (Balance, Balance, Balance) {
+		//execute the issue
+		let issue_amount_1 = 3000;
+		let issue_fee_1 = 1000;
+		let griefing_collateral_1 = 1;
+		let amount_transferred_1 = 3000;
+		let issue_id_1 =
+			setup_execute(issue_amount_1, issue_fee_1, griefing_collateral_1, amount_transferred_1)
+				.unwrap();
+		assert_ok!(execute_issue(USER, &issue_id_1));
+
+		//execute the issue on the other currency
+		let issue_amount_2 = 3000;
+		let issue_fee_2 = 500;
+		let griefing_collateral_2 = 1;
+		let amount_transferred_2 = 3000;
+		let issue_id_2 = setup_execute_with_vault(
+			issue_amount_2,
+			issue_fee_2,
+			griefing_collateral_2,
+			amount_transferred_2,
+			VAULT_2.clone(),
+		)
+		.unwrap();
+		assert_ok!(execute_issue(USER, &issue_id_2));
+
+		let issue_amount_3 = 3000;
+		let issue_fee_3 = 600;
+		let griefing_collateral_3 = 1;
+		let amount_transferred_3 = 3000;
+		let issue_id_3 = setup_execute_with_vault(
+			issue_amount_3,
+			issue_fee_3,
+			griefing_collateral_3,
+			amount_transferred_3,
+			VAULT_3.clone(),
+		)
+		.unwrap();
+		assert_ok!(execute_issue(USER, &issue_id_3));
+
+		let issue_amount_4 = 3000;
+		let issue_fee_4 = 400;
+		let griefing_collateral_4 = 1;
+		let amount_transferred_4 = 3000;
+		let issue_id_4 = setup_execute_with_vault(
+			issue_amount_4,
+			issue_fee_4,
+			griefing_collateral_4,
+			amount_transferred_4,
+			VAULT_4.clone(),
+		)
+		.unwrap();
+		assert_ok!(execute_issue(USER, &issue_id_4));
+
+		let issue_amount_5 = 3000;
+		let issue_fee_5 = 700;
+		let griefing_collateral_5 = 1;
+		let amount_transferred_5 = 3000;
+		let issue_id_5 = setup_execute_with_vault(
+			issue_amount_5,
+			issue_fee_5,
+			griefing_collateral_5,
+			amount_transferred_5,
+			VAULT_5.clone(),
+		)
+		.unwrap();
+		assert_ok!(execute_issue(USER, &issue_id_5));
+
+		return (issue_fee_1 + issue_fee_4, issue_fee_2 + issue_fee_5, issue_fee_3)
```

Comment:
This is a little confusing. When looking at the mock.rs file I can see that eg. vaults 1 and 4 use the same wrapped currency and so do vault 2 and 5. But it's not clear here.
Let's either add a comment here or at the top of this function so that a reader knows why these values are summed up and what they represent.